### PR TITLE
added usbc.json with pd controller draft

### PR DESCRIPTION
**Pull request by rsnancollas**

### CODE REVIEW

File: part-spec/usbc.json [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+                    ],
+                    "type": "string"
+                },
+                "usbTypecRevision": {
+                    "description": "usb type-c spec revision implemented by controller",
+                    "examples": [
+                         "1.2",
+                         "1.3"
+                    ],
+                    "type": "number"
```
Comment (asadate) [inline part-spec/usbc.json]: Change "type" as "string"

File: part-spec/usbc.json [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+                    "description": "roles supported by pd controller",
+                    "examples": [
+                        "Source",
+                        "Sink",
+                        "DRP"
+                     ],
+                     "type": "string"
+                },
+                "fastRoleSwapSupport": {
+                    "description": "whether the pd controller supports fast rope swap (FRS)",
```
Comment (asadate) [inline part-spec/usbc.json]: typo: "role" instead of "rope"

File: part-spec/usbc.json [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+                    "$ref": "definitions.json#/unit"
+                },
+                "vconnMaxCurrent": {
+                    "description": "maximum continuous current supported by internal vconn switch (if supported)", 
+                    "comment": "units of amps",
+                    "$ref": "definitions.json#/unit"
+                },
+                "vconnOverCurrentLimit": {
+                    "description": "over current limit supported by internal vconn switch (if supported)",
+                    "comment": "units of ampss",
```
Comment (asadate) [inline part-spec/usbc.json]: typo: "amps"

File: part-spec/usbc.json [hunk 4]
```diff
@@ -0,0 +1,184 @@
...
+            "properties": {
+                "componentID": {
+                    "description": "common component identifying information, such as mpn",
+                    "$ref": "definitions.json#/componentID"
+                },
+                "pdVersion": {
+                    "description": "version of power delivery spec implemented by controller",
+                    "examples": [
+                        "pd2",
+                        "pd3"
```
Comment (DianaZig) [inline part-spec/usbc.json]: Should the examples detail both the revision and version, given that the PD 3.1 spec introduced the Revision message which will report both?

File: part-spec/usbc.json [hunk 5]
```diff
@@ -0,0 +1,184 @@
...
+                },
+                "usbTypecRevision": {
+                    "description": "usb type-c spec revision implemented by controller",
+                    "examples": [
+                         "1.2",
+                         "1.3"
+                    ],
+                    "type": "string"
+                },
+                "roleSupported": {
```
Comment (DianaZig) [inline part-spec/usbc.json]: powerRoleSupported?  Just to disambiguate from any data role support

File: part-spec/usbc.json [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+                    "description": "roles supported by pd controller",
+                    "examples": [
+                        "Source",
+                        "Sink",
+                        "DRP"
+                     ],
+                     "type": "string"
+                },
+                "fastRoleSwapSupport": {
+                    "description": "whether the pd controller supports fast role swap (FRS)",
```
Comment (DianaZig) [inline part-spec/usbc.json:45]: Would this need to be broken out as FRS support for initial sink versus initial source?  Or is the assumption that a part supporting one will also support the other?

File: part-spec/usbc.json (hunk 4 above)
Comment (asadate) [reply part-spec/usbc.json]: Good point. Will change to PD2.0 and PD3.1 as examples

File: part-spec/usbc.json (hunk 6 above)
Comment (asadate) [reply part-spec/usbc.json:45]: The specs call out FRS but I haven't seen that broken into sink to source switch vs source to sink switch.

File: part-spec/usbc.json (hunk 5 above)
Comment (asadate) [reply part-spec/usbc.json]: Noted. Will change

**Review by DianaZig — APPROVED** [review]
Looks good

**Review by Carrotman42 — APPROVED** [review]
Approving on behalf of Diana (since she doesn't have Write access, GitHub didn't care about her approval apparently)

**Pull request merged**